components/setup/registry-setup.tsx

```diff
@@ -147,6 +148,9 @@ export function RegistrySetup({
         return setChainRegistry(contractAddress);
       })
       .then(() => {
+        const chain = supportedChains?.find(({chainId}) => chainId === +connectedChain?.id);
```

Comment:
We have `useChain` hook with the method `findSupportedChain`, I think you could use it here.
https://github.com/bepronetwork/web-network/blob/dev/x-hooks/use-chain.ts#L21